flybywiresim/aircraft
# feat(bleed): crossbleed starts and more

Fixes #820

## Summary of Changes

Detailed list of changes:
- Allow engine start from crossbleed air
- Implement pressure sensors for regulated, transfer, and precooler inlet pressure
- Implement precooler outlet temperature sensor
- Implement fully open/fully closed sensors for the crossbleed valve.
- Update SD Bleed page indications
  - Implement crossbleed valve indications for open, closed and in-transfer status
  - Precooler inlet pressure green/amber logic adjusted
  - Use 5s confirmation before showing closed pressure regulating valve indication
- Add bleed air overpressure/underpressure/overheat detection in the BMC (no ECAM warnings yet)
- Improve fan air valve regulation logic
- Rework engine compressor physics
- Implement overpressure valve
- Rework high pressure valve regulation
- Increase precooler cooling efficiency
- Tune APU bleed pressure
- Rework pack fault light logic

## Screenshots (if necessary)



## References





## Additional context



Discord username (if different from GitHub): BlueberryKing

## Testing instructions

- Load up a flight
- Make sure starting an engine from APU bleed works
- Make sure performing a crossbleed start works
  - Turn off APU Bleed off
  - Turn the engine bleed pushbutton off on the receiving engine
  - Open the crossbleed valve
  - Set the started engine to about 30% N1
  - Set the engine mode selector to IGN
  - Set the engine master switch to ON
- Fly in various conditions, make sure indications on the BLEED SD page, stay green


## How to download the PR for QA

Every new commit to this PR will cause a new A32NX artifact to be created, built, and uploaded.

1. Make sure you are signed in to GitHub
1. Click on the **Checks** tab on the PR
1. On the left side, click on the bottom **PR** tab
1. Click on the **A32NX** download link at the bottom of the page
